bindings.go

```diff
@@ -0,0 +1,181 @@
+package libovsdb
+
+import (
+	"fmt"
+	"reflect"
+)
+
+var (
+	intType  = reflect.TypeOf(0)
+	realType = reflect.TypeOf(0.0)
+	boolType = reflect.TypeOf(true)
+	strType  = reflect.TypeOf("")
+	nilType  = reflect.TypeOf(nil)
+)
+
+// nativeTypeFromExtended returns the native type that can hold a value of an
+// Extended type
+func nativeTypeFromExtended(extended ExtendedType) (reflect.Type, error) {
+	switch extended {
+	case TypeInteger:
+		return intType, nil
+	case TypeReal:
+		return realType, nil
+	case TypeBoolean:
+		return boolType, nil
+	case TypeString:
+		return strType, nil
+	case TypeUUID:
+		return strType, nil
+	default:
+		return nilType, fmt.Errorf("Failed to determine type of column %s", extended)
```

Comment:
I don't think panic()ing in library code is right. It bypasses any attempt from the main program to recover from the error or properly clean before exit.

Reply:
For my understanding panic is the way we do assertion in Go (Please correct me if I am wrong). In the above switch-case scenario it is determined by code in this library itself that the "default" branch is impossible, so "assertion" should be used instead of error handling. Error handling and logging should be used for handling errors related to external input. I think it is proper to use assertion in library code. It helps make the code clean (while not sacrificing reliability). As we can see this function doesn't need error return and it would in turn simplify the callers' logic. If there is a better way to do assertion in Go I am open to it.

Reply:
Sorry that I need to adjust my comments here. I still insist the principle I mentioned above, but for this case it is currently depending on external input: the "type" being passed in to the function nativeTypeFromExtended() eventually comes from parsing a schema. So if we consider only this function now, we should do error handling instead of panic. However, my proposal is that we shouldn't let the error float this far. We should validate the schema when parsing it and make sure we never store any unknown type in the column schema data structure. See my comment at [0]. This way, we will be sure when calling nativeTypeFromExtended() there is never going to be any unknown types passed in, and assertion (panic) is the proper approach which simplifies the code.

I totally agree that in library code we shouldn't use panic for error handling, but I just want to emphasize that here we don't need error handling - we should move the error handling to the right place - when parsing external schema.

[0] 7d3749a#diff-c3b0d951288715c695f51ce2727c87a056e89ccbb4f3003be71baa8b405383e9R292

Reply:
Agree. I'll update the PR handling the possible errors at the schema-parsing level. Thanks @hzhou8